.circleci/config.yml

```diff
@@ -76,6 +77,7 @@ jobs:
 
       - run:
           name: build docs
+          no_output_timeout: 20m
```

Comment:
This is actually indicative of a problem I think. I'm just trying to build locally, and Sphinx is attempting to melt my laptop. And my doc build fix PR gh-13803 is timing out even with this setting.

Do you have an idea what is causing the slowdown @tupui?

Reply:
It wasn't this PR - I just timed pre/post the merge commit of this PR, and timing were almost identical. Still curious why the timeout needed to be included here.

On my laptop, `make html` takes about 20 minutes as well.

Reply:
I am also needing around 20 minutes to build the doc. But on my side I think it was always the case (I mean January here 😅 as before I did not remember trying to do it locally). I guess it's due to all the examples and doctests we have. I am not sure if there are some tools from Sphinx (or extensions) to time the doc build.

Reply:
Another thing to check could be the sphinx version (I know that with the pandas docs at some point it took longer after a sphinx update). 

Now, it could certainly be that this theme give *some* slowdown, because we do some manipulation of the navigation html for each page. If that's the case, it would be good to get some timings of this to see if there are things we should try to optimize.